src/sasctl/pzmm/write_json_files.py

```diff
@@ -950,3 +982,629 @@ def convertDataRole(self, dataRole):
+                        }
+                    ],
+                    indent=4,
+                )
+                file.write(json_step)
+
+            for package, version in package_and_version:
+                if version:
+                    json_step = json.dumps(
+                        [
+                            {
+                                "step": "install " + package,
+                                "command": "pip install " + package + "==" + version,
+                            }
+                        ],
+                        indent=4,
+                    )
+                file.write(json_step)
+
+    @classmethod
+    def get_local_package_version(cls, package_list):
+        """
+        Get package_name versions from the local environment. If the package_name
+        does not contain an attribute of "__version__", "version", or
+        "VERSION", no package_name version will be found.
+
+        Parameters
+        ----------
+        package_list : list
+            List of Python packages.
+
+        Returns
+        -------
+        list
+            Nested list of Python package_name names and found versions.
+        """
+
+        def package_not_found_output(package_name, package_versions):
+            print(
+                f"Warning: Package {package_name} was not found in the local environment. Either {package_name} is not "
+                f"a valid Python package, or the package is not present in this environment. The requirements.json file"
+                f" will include a commented out version of the pip installation command at the bottom of the file. "
+                f"Please review the file and verify that the package exists and input the version needed."
+            )
+            package_versions.append([package_name, None])
+            return package_versions
+
+        package_and_version = []
+        import importlib
+
+        for package in package_list:
+            try:
+                name = importlib.import_module(package)
+                try:
+                    package_and_version.append([package, name.__version__])
+                except AttributeError:
+                    try:
+                        package_and_version.append([package, name.version])
+                    except AttributeError:
+                        try:
+                            package_and_version.append([package, name.VERSION])
+                        except AttributeError:
+                            package_and_version = package_not_found_output(
+                                package, package_and_version
+                            )
+            except ModuleNotFoundError:
+                package_and_version = package_not_found_output(
+                    package, package_and_version
+                )
+
+        return package_and_version
+
+    @classmethod
+    def get_code_dependencies(cls, json_path=Path.cwd()):
```

Comment:
Would not call this `json_path`.  There's no JSON involved in this method, nor does the docstring reference JSON.

Reply:
I think model_path probably makes more sense then, as the aim is to get the user pointing at the directory their model is in.